exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -248,9 +244,30 @@ func newLogsExporter(config Config, set exporter.CreateSettings, marshalers map[
 
 	return &kafkaLogsProducer{
 		cfg:       config,
-		topic:     config.Topic,
 		marshaler: marshaler,
 		logger:    set.Logger,
 	}, nil
 
 }
+
+type resourceSlice[T any] interface {
+	Len() int
+	At(int) T
+}
+
+type resource interface {
+	Resource() pcommon.Resource
+}
+
+func getTopic[T resource](cfg *Config, resources resourceSlice[T]) string {
+	if cfg.TopicFromAttribute == "" {
+		return cfg.Topic
+	}
+	for i := 0; i < resources.Len(); i++ {
+		rv, ok := resources.At(i).Resource().Attributes().Get(cfg.TopicFromAttribute)
+		if ok && rv.Str() != "" {
+			return rv.Str()
+		}
+	}
+	return cfg.Topic
```

Comment:
We should document that if `TopicFromAttribute` is set, but not found, `Topic` will be used.

Reply:
Thank you for the feedback. I've made the addition.

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31809/commits/44272fa55f0b9bb3040a84b413147950fa1d9645